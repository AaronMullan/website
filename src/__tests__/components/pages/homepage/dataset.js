import React from 'react'
import renderer from 'react-test-renderer'
import Dataset from '~components/pages/homepage/dataset'
import { useStaticQuery } from 'gatsby'

const mockImage = {
  aspectRatio: 1.3,
  sizes: '(max-width: 1200px) 100vw, 1200px',
  src:
    '/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png',
  srcSet: '/srcset',
  tracedSVG: '[SVG]',
}

describe('Components : Pages : Homepage : Dataset', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Dataset title='example title' image={ mockImage } imageAlt='alt text'>
        This is a summary of a source that uses our data
      </Dataset>
    ).toJSON()
    expect(tree).toMatchSnapshot()

    const treeFlipped = renderer.create(
      <Dataset title='example title' image={ mockImage } imageAlt='alt text' flip>
        This is a flipped summary of a source that uses our data
      </Dataset>
    ).toJSON()
    expect(treeFlipped).toMatchSnapshot()
  })
})
