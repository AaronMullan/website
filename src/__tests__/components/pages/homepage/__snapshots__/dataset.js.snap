// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Homepage : Dataset renders correctly 1`] = `
<div
  className="dataset "
>
  <div
    className="image"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "76.92307692307692%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        aria-hidden={true}
        src="[SVG]"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 1200px) 100vw, 1200px"
          srcSet="/srcset"
        />
        <img
          alt="alt text"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 1200px) 100vw, 1200px"
          src="/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png"
          srcSet="/srcset"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/srcset\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" srcset=\\"/srcset\\" src=\\"/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png\\" alt=\\"alt text\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
  <div
    className="info"
  >
    <h3>
      example title
    </h3>
    <p>
      This is a summary of a source that uses our data
    </p>
  </div>
</div>
`;

exports[`Components : Pages : Homepage : Dataset renders correctly 2`] = `
<div
  className="dataset flip"
>
  <div
    className="image"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "76.92307692307692%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        aria-hidden={true}
        src="[SVG]"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 1200px) 100vw, 1200px"
          srcSet="/srcset"
        />
        <img
          alt="alt text"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 1200px) 100vw, 1200px"
          src="/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png"
          srcSet="/srcset"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/srcset\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" srcset=\\"/srcset\\" src=\\"/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png\\" alt=\\"alt text\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
  <div
    className="info"
  >
    <h3>
      example title
    </h3>
    <p>
      This is a flipped summary of a source that uses our data
    </p>
  </div>
</div>
`;
